docs/scos/dev/upgradability-services/new-documentation/method-of-extended-class-overridden-on-project-level.md

```diff
@@ -0,0 +1,105 @@
+---
+title: Method of extended class is overridden one the project level
+description: Reference information for evaluator and upgrader tools.
+last_updated: Mar 23, 2022
+template: concept-topic-template
+---
+
+## Method of an extended class is overridden on the project level
+
+If you extend functionality under private API, any update can cause errors or unexpected changes in functionality.
+If you extend a core class and override one of its methods, updates can cause errors or unexpected changes in functionality.
+
+More information about private API [see here]{link to Private API class was extended or used - private-api-class-extended-or-used.md#Private API class was extended or used}
+
+#### Using custom methods on the project level
+
+To avoid unexpected issues and achieve the same result, instead of overriding the core methods, introduce custom ones.
+
+{% info_block infoBox "" %}
+
+While public API updates always support backward compatibility, private API updates can break backward compatibility. So, backward compatibility is not guaranteed in the private API. For example, if you use a core method on the project level, and it is updated or removed, it can cause unexpected issues during updates.
+
+{% endinfo_block %}
+
+#### Example of code that can cause upgradability errors
+
+For example, the extended class EvaluatorCategoryImageEntityManager overrides the core method CategoryImageEntityManager.
+
+```php
+namespace Pyz\Zed\Evaluator\Persistence;
+
+use Generated\Shared\Transfer\CategoryImageSetTransfer;
+use Spryker\Zed\CategoryImage\Persistence\CategoryImageEntityManager;
+
+class EvaluatorCategoryImageEntityManager extends CategoryImageEntityManager
+{
+    /**
+     * @param \Generated\Shared\Transfer\CategoryImageSetTransfer $categoryImageSetTransfer
+     *
+     * @return \Generated\Shared\Transfer\CategoryImageSetTransfer
+     */
+    public function saveCategoryImageSet(CategoryImageSetTransfer $categoryImageSetTransfer): CategoryImageSetTransfer
```

Comment:
An example is not clear, as we use extended class functionality, e.g via `$a = parent::saveCategoryImageSet()`

Reply:
Would you like to have the whole example with using the 
EvaluatorCategoryImageEntityManager::saveCategoryImageSet?

or 

Would you like to update the saveCategoryImageSet method itself to add a parent usage there? 

Both cases are not needed from my perspective. 
But lets discuss. Seems liek we have a different opinion on it :)

Reply:
For now we can skip it but will not resolve.